edkrepo_manifest_parser/edk_manifest.py

```diff
@@ -582,6 +582,8 @@ def generate_pin_etree(self, description, combo_name, repo_source_list):
         project_root.find('Description').text = description
 
         subroot_m = self._tree.find('GeneralConfig')
+        current_cloned_combo = subroot_m.find('CurrentClonedCombo')

```

Comment:
we should add a check to ensure that the combo name is not None ... in that case we should retain the existing current cloned combo value

Reply:
Changing this to an input check on combo_name to ensure it's in the source manifest file. Error out otherwise